includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -798,6 +799,31 @@ public function detach_payment_method( $payment_method_id ) {
 		);
 	}
 
+	/**
+	 * Records a new Terms of Service agreement.
+	 *
+	 * @param string $source     A string, which describes where the merchant agreed to the terms.
+	 * @param string $user_name  The user_login of the current user.
+	 * @param string $user_ip    IP address of the current user.
+	 * @param string $user_agent User agent string for the current user.
+	 *
+	 * @return bool A flag, which indicates whether the agreement was saved.
+	 *
+	 * @throws API_Exception If an error occurs.
+	 */
+	public function add_tos_agreement( $source, $user_name, $user_ip, $user_agent ) {
+		return 'success' === $this->request(
```

Comment:
1. `$request` returns an array so the comparison should always be `false`.
2. Most of the other api_client methods return array. Any reason not to do so here?
3. I can't find the place where return result would be used, so I don't fully understand the initial idea.

Reply:
You are right. I removed the comparison [here](https://github.com/Automattic/woocommerce-payments/pull/993/commits/188063824128a91d03a15f402903dc28015e78d9), because the return value is not really used, relying on exceptions instead.